Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift

```diff
@@ -519,6 +521,24 @@ internal class MiniAppScriptMessageHandler: NSObject, WKScriptMessageHandler {
         )
     }
 
+    func downloadFile(with callbackId: String, parameters: RequestParameters?) {
+         if
+            isUserAllowedPermission(customPermissionType: .fileDownload, callbackId: callbackId),
+            let fileName = parameters?.filename,
+            let url = parameters?.url,
+            let headers = parameters?.headers
+        {
+             hostAppMessageDelegate?.downloadFile(fileName: fileName, url: url, headers: headers) { (result) in
+                switch result {
+                case .success:
+                    self.executeJavaScriptCallback(responseStatus: .onSuccess, messageId: callbackId, response: fileName)
+                case .failure:
+                    self.executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: getMiniAppErrorMessage(MiniAppJavaScriptError.unexpectedMessageFormat))
```

Comment:
I think we need to use a specific format for the Error, similar to what we did with `getAccessToken`. Also in the case that the download failed, we should pass the status code and message. This is what I suggested, but you can sync up with Android side:
```
{
  status: number,
  reason: String
}
```

Reply:
added a new error `MASDKDownloadFileError` that should cover the error cases